pkg/api/v1/clients/kube/cache/cache.go

```diff
@@ -52,7 +52,7 @@ func NewKubeCoreCache(ctx context.Context, client kubernetes.Interface) (*KubeCo
 	}
 
 	kubeController := controller.NewController("kube-plugin-controller",
-		controller.NewLockingSyncHandler(k.updatedOccured),
+		controller.NewLockingSyncHandler(k.updatedOccured), false,
 		pods.Informer(), configMaps.Informer(), secrets.Informer())
 
 	stop := ctx.Done()
```

Comment:
Instead of adding that flag, can't we just remove the `go kubeInformerFactory.Start(stop)` line? If you look at that function, it just starts all the informers in the factory: 
```go
// Start initializes all requested informers.
func (f *sharedInformerFactory) Start(stopCh <-chan struct{}) {
	f.lock.Lock()
	defer f.lock.Unlock()

	for informerType, informer := range f.informers {
		if !f.startedInformers[informerType] {
			go informer.Run(stopCh)
			f.startedInformers[informerType] = true
		}
	}
}
```
That is the same thing that our kube controller would do if you passed it the same informers, which we do. 

Basically, it seems to me that all we need the `kubeInformerFactory` for is initializing the core informers via those `kubeInformerFactory.Core().V1().*` functions.

Reply:
that is a good idea; i didnt noticed that the sharedInformerFactory isnt really used last night.
i saw the code now and indeed the only thing we do is call `Start()` so we can just not do that.

Reply:
you can skip calling `kubeInformerFactory.Start` but still call the `kubeController.WaitForInformers` I added in this PR to make it more foolproof (instead of getting the list of informers again)